src/nodes.coffee

```diff
@@ -4803,20 +4803,46 @@ makeDelimitedLiteral = (body, options = {}) ->
 lesser  = (a, b) -> if a < b then a else b
 greater = (a, b) -> if a > b then a else b
 
+isAstLocGreater = (a, b) ->
+  return yes if a.line > b.line
+  return no unless a.line is b.line
+  a.column > b.column
+
+isLocationDataStartGreater = (a, b) ->
+  return yes if a.first_line > b.first_line
+  return no unless a.first_line is b.first_line
+  a.first_column > b.first_column
+
+isLocationDataEndGreater = (a, b) ->
+  return yes if a.last_line > b.last_line
+  return no unless a.last_line is b.last_line
+  a.last_column > b.last_column
+
 # Take two nodes’ location data and return a new `locationData` object that
 # encompasses the location data of both nodes. So the new `first_line` value
 # will be the earlier of the two nodes’ `first_line` values, the new
 # `last_column` the later of the two nodes’ `last_column` values, etc.
 mergeLocationData = (locationDataA, locationDataB) ->
-  return
-    first_line:   lesser locationDataA.first_line,   locationDataB.first_line
```

Comment:
This logic was buggy (in `mergeLocationData()` and similarly in `mergeAstLocationData()`) - we can't individually compare line and column, we need to treat line/column as a single object for comparison (eg line 4 column 0 is greater than line 3 column 2, so we want to end up with line 4 column 0, not line 4 column 2)